x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/TextExpansionQueryBuilder.java

```diff
@@ -242,6 +251,9 @@ protected int doHashCode() {
     }
 
     public static TextExpansionQueryBuilder fromXContent(XContentParser parser) throws IOException {
+
+        deprecationLogger.warn(DeprecationCategory.API, NAME, TEXT_EXPANSION_DEPRECATION_MESSAGE);
```

Comment:
Just curious, why is this logged at `warn` level here, but at `critical` level for `weighted_tokens`?

Reply:
Good question! 

If you look at the [DeprecationLogger](https://github.com/elastic/elasticsearch/blob/1c1733d82338fa75f2780f0a610d11dba4ab61a5/server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java#L36) you'll see the following: 

- Critical logs indicate that a change will break in next version. 
- Warn logs indicate that it's a less critical deprecation that won't break in the next version. 

Since `text_expansion` is GA it will not be removed until 9.0. However, `weighted_tokens` is tech preview and thus is slated for removal in 8.16.